src/sim/trace-extract/google_trace_simulator.cc

```diff
@@ -262,62 +443,68 @@ void GoogleTraceSimulator::AddTaskStats(
   TaskStats* task_stats = FindOrNull(task_id_to_stats_, task_identifier);
   if (task_stats == NULL) {
     // Already added stats.
+    VLOG(2) << "Skipping setting runtime for "
+            << task_identifier.job_id << "/" << task_identifier.task_index;
     return;
   }
   TaskDescriptor* td_ptr = FindPtrOrNull(task_id_to_td_, task_identifier);
   CHECK_NOTNULL(td_ptr);
   uint64_t* task_runtime_ptr = FindOrNull(*task_runtime, task_identifier);
-  uint64_t runtime = 0;
+  double runtime = 0.0;
   if (task_runtime_ptr != NULL) {
-    runtime = *task_runtime_ptr;
+    // knowledge base has runtime in ms, but we get it in micros
+    runtime = *task_runtime_ptr / 1000.0;
   } else {
     // We don't have information about this task's runtime.
     // Set its runtime to max which means it's a service task.
+    // TODO(adam): or float infinity?
     runtime = numeric_limits<uint64_t>::max();
   }
-  vector<EquivClass_t>* equiv_classes =
-    cost_model_->GetTaskEquivClasses(td_ptr->uid());
-  // XXX(ionel): This assumes that we have one task equivalence class per task.
-  for (vector<EquivClass_t>::iterator it = equiv_classes->begin();
-       it != equiv_classes->end(); ++it) {
-    knowledge_base_->SetAvgRuntimeForTEC(*it, runtime);
-    if (fabs(task_stats->avg_mean_cpu_usage + 1.0) > EPS) {
-      knowledge_base_->SetAvgMeanCpuUsage(*it, task_stats->avg_mean_cpu_usage);
-    }
-    if (fabs(task_stats->avg_canonical_mem_usage + 1.0) > EPS) {
-      knowledge_base_->SetAvgCanonicalMemUsage(
-          *it, task_stats->avg_canonical_mem_usage);
-    }
-    if (fabs(task_stats->avg_assigned_mem_usage + 1.0) > EPS) {
-      knowledge_base_->SetAvgAssignedMemUsage(
-          *it, task_stats->avg_assigned_mem_usage);
-    }
-    if (fabs(task_stats->avg_unmapped_page_cache + 1.0) > EPS) {
-      knowledge_base_->SetAvgUnmappedPageCache(
-          *it, task_stats->avg_unmapped_page_cache);
-    }
-    if (fabs(task_stats->avg_total_page_cache + 1.0) > EPS) {
-      knowledge_base_->SetAvgTotalPageCache(
-          *it, task_stats->avg_total_page_cache);
-    }
-    if (fabs(task_stats->avg_mean_disk_io_time + 1.0) > EPS) {
-      knowledge_base_->SetAvgMeanDiskIOTime(
-          *it, task_stats->avg_mean_disk_io_time);
-    }
-    if (fabs(task_stats->avg_mean_local_disk_used + 1.0) > EPS) {
-      knowledge_base_->SetAvgMeanLocalDiskUsed(
-          *it, task_stats->avg_mean_local_disk_used);
-    }
-    if (fabs(task_stats->avg_cpi + 1.0) > EPS) {
-      knowledge_base_->SetAvgCPIForTEC(*it, task_stats->avg_cpi);
-    }
-    if (fabs(task_stats->avg_mai + 1.0) > EPS) {
-      knowledge_base_->SetAvgIPMAForTEC(*it, 1.0 / task_stats->avg_mai);
-    }
+  EquivClass_t bogus_equiv_class = (EquivClass_t)td_ptr->uid();
```

Comment:
I've added a comment, but it's not as you assume: by contrast, each task has its own, single-entity equivalence class. I think Adam did this as a short-cut hack in order to have information in the KB even though the SimulatedQuincyCostModel does not actually have a proper notion of TECs (it would record the stats on a per-rack basis!). 

We almost certainly need to do some work here to improve statistics collection. Okay to merge anyway with my changes?